receiver/azureeventhubreceiver/README.md

```diff
@@ -24,18 +27,67 @@ The offset at which to start watching the event hub. If empty, it starts with th
 
 Default: ""
 
-Example:
+### format (Optional)
+Determines how to transform the Event Hub messages into OpenTelemetry logs. See the "Format"
+section below for details.
+
+Default: "raw"
+
+### Example Configuration
 
 ```yaml
 receivers:
   azureeventhub:
     connection: Endpoint=sb://namespace.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=superSecret1234=;EntityPath=hubName
     partition: foo
     offset: "1234-5566"
+    format: "azure"
 ```
 
 This component can persist its state using the [storage extension].
 
+## Format
+
+### raw
+
+The "raw" format maps the AMQP properties and data into the
+attributes and body of an OpenTelemetry LogRecord, respectively.
+The body is represented as a raw byte array.
+
+### azure
+
+The "azure" format extracts the Azure log records from the AMQP
+message data, parses them, and maps the fields to OpenTelemetry
+attributes. The table below summarizes the mapping between the 
+[Azure common log format](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema)
+and the OpenTelemetry attributes.
+
+
+| Azure                            | OpenTelemetry                          | 
+|----------------------------------|----------------------------------------|
+| callerIpAddress (optional)       | net.sock.peer.addr (attribute)         | 
+| correlationId (optional)         | azure.correlation.id (attribute)       | 
+| category (optional)              | azure.category (attribute)             | 
+| durationMs (optional)            | azure.duration (attribute)             | 
+| Level (optional)                 | severity_number, severity_text (field) | 
+| location (optional)              | cloud.region (attribute)               | 
+| —                                | cloud.provider (attribute)             | 
+| operationName (required)         | azure.operation.name (attribute)       |
+| operationVersion (optional)      | azure.operation.version (attribute)    | 
+| properties (optional)            | azure.properties (attribute, nested)   | 
+| resourceId (required)            | azure.resource.id (resource attribute) | 
+| resultDescription (optional)     | azure.result.description (attribute)   | 
+| resultSignature (optional)       | azure.result.signature (attribute)     | 
+| resultType (optional)            | azure.result.type (attribute)          | 
+| tenantId (required, tenant logs) | azure.tenant.id (attribute)            | 
+| time (required)                  | time_unix_nano (field)                 | 
+| identity (optional)              | azure.identity (attribute, nested)     |
+
+Note: JSON does not distinguish between fixed and floating point numbers. This
+formatter will first try to parse a number as an integer, if that works, then
+an integer is used for the OpenTelemetry attribute. In all other cases, a double
+is used.
```

Comment:
I'm not convinced this is a good idea. Isn't it possible that this will lead to some attributes that switch data types depending on the values? In my opinion, it's ok to parse as int or float, but it should be predefined for each attribute and based on understanding of the value.

Reply:
The problem is that only the top-level schema is fixed. The  `properties` attribute uses a schema that depends on the specific Azure service/resource. I've not found a comprehensive list of those attributes and even if it existed, keeping this synchronized with upstream changes would be a challenge.
The only two implementations that seem reasonable to me are:
1. Use a heuristic to determine the type of JSON number attributes (current implementation) or
2. Default to using doubles for all JSON number attributes.
I'm OK with either. @djaglowski Would you prefer the second option?

Reply:
In that case, I think the second option is preferable. Better to have guaranteed stability in the value type.

Reply:
I've implemented #2.